src/app/core/utils/url-resource-ids.ts

```diff
@@ -0,0 +1,3 @@
+export function encodeResourceID(object: string): string {
+  return encodeURIComponent(encodeURIComponent(object));
+}
```

Comment:
This would be the place to write a comment that `encodeResourceID` currently means "double encoding" and a short explanation why?

Reply:
I would actually implement this as a HttpInterceptor, to make sure it is used everywhere.

Reply:
I really think this is an ICM issue and not a PWA issue 🤔

Reply:
We had a very close look into this, and with the email addresses being used currently in the requests, it cannot be solved within the ICM. There is a team looking into using other IDs (without special characters), but this will take time and we will still need a fix for the email addresses for legacy reasons. I will look into HttpInterceptors though.

Reply:
You can also introduce an interceptor that always does `return next.handle(req.clone({ url: encodeURI(req.url) }));`. Works also for other cases. 😉

Reply:
The http interceptor seemed very promising. Unfortunatelly this does not work. It either encodes not enough (encodeURI does not encode a lot of characters that we need encoded) or not enough (encodeURIComponent does encode everything if called with the whole URI). 
Therefore it still, sadly, seems like the best solution to directly encode the specific resource ids.

Reply:
@DiverDori So as far as I can see for now you only want to double encode the email part in the URL, correct? This can still be done with an interceptor. Let me try.

Reply:
```
@Injectable({ providedIn: 'root' })
export class EncodeEmailInterceptor implements HttpInterceptor {
  intercept(req: HttpRequest<unknown>, next: HttpHandler) {
    if (decodeURIComponent(req.url).includes('@') && !req.url.includes('/quotes/')) {
      const regex = /^.*\/(.*?@.*?)\/.*$/;
      const match = regex.exec(decodeURIComponent(req.url));

      const email = match?.[1];
      if (email) {
        const newURL = decodeURIComponent(req.url).replace(email, encodeURIComponent(encodeURIComponent(email)));
        return next.handle(req.clone({ url: newURL }));
      }
    }
    return next.handle(req);
  }
}
```
That would be just replacing the email. But I can see that quoting is no longer working for standard users. So is it even correct that you double-encoded the quoting related services in this PR?

Reply:
If this was in Java, I would decline. This is effort (decode and includes()) in every case, without being able to handle everything (like starting to include customerID to be encoded twice). 
The fix for quotes and quoterequests is in ICM and not yet released. I have to talk to the PWA team how this is usually handled.

Reply:
Yes, this should be restricted to `PLATFORM_SERVER` for sure, you're right! But apart from that I don't see an issue delegating this extra effort to all client applications. 😃

Reply:
We will use the `encodeResourceID` method for the places needed.
If a project still needs the former way of handling the encoding it is just this one place in this method where the project can replace the double encoding with a single encoding.